sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,146 @@
+        -------
+        :obj:`pandas.Series` or `pandas.DataFrame`
+                returns: For each column/row the number of non-NA/null entries.\
+                     If level is specified returns a DataFrame.
+        """
+
+        _func_name = 'Method pandas.dataframe.count().'
+
+        if not isinstance(self, DataFrameType):
+            raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
+
+        if not (isinstance(axis, types.Omitted) or axis == 0):
+            raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
+            raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+
+        def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
+            result_data = []
+            result_index = []
+
+            for dataframe_item in self._data:
+                item_count = dataframe_item.count()
+                item_name = dataframe_item._name
+                result_data.append(item_count)
+                result_index.append(item_name)
+
+            return pandas.Series(data=result_data, index=result_index)
+
+        return sdc_pandas_dataframe_count_impl
+
+else:
+    def sdc_pandas_dataframe_reduce_columns(df, name, param):
+        saved_columns = df.columns
+        n_cols = len(saved_columns)
+        data_args = tuple('data{}'.format(i) for i in range(n_cols))
+        help_param = ', {}={}):'
+        func_text = 'def _reduce_impl(df):'
+        all_params = ['df']
+        for name, value in param:
+            all_params.append('{}={}'.format(name, value))
+        func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
+        func_lines = [func_definition]
+        for i, d in enumerate(data_args):
+            line = '  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+            func_lines.append(line.format(d + '_S', i))
+            func_lines.append('  {} = {}.{}()'.format(d + '_O', d + '_S', name))
+        func_lines.append('  data = np.array(({},))'.format(
+            ", ".join(d + '_O' for d in data_args)))
+        func_lines.append('  index = hpat.str_arr_ext.StringArray(({},))'.format(
+            ', '.join('"{}"'.format(c) for c in saved_columns)))
+        func_lines.append('  return hpat.hiframes.api.init_series(data, index)')
+        loc_vars = {}
+        func_text = '\n'.join(func_lines)
+
+        exec(func_text, {'hpat': sdc, 'np': numpy}, loc_vars)
+        _reduce_impl = loc_vars['_reduce_impl']
+
+        return _reduce_impl
+
+    def check_type(name, df, axis=None, skipna=None, level=None, numeric_only=None, ddof=1, min_count=0):
```

Comment:
Is it a common function for checking all parameters? How is it related to df.head()?